### PR TITLE
V2

### PR DESCRIPTION
**Pull request by valivia**
This pull request implements a completely new custom system for managing the state of the music bot, A large chunk of this code is derived from the original RavenBot music bot code.
This has been done because the package used previously was highly unreliable and hard to maintain

### CODE REVIEW

**Review by luksab — APPROVED** [review]
You've improved your code style a lot.
Great work!

File: src/lib/queueInfo.ts [hunk 1]
```diff
@@ -1,26 +1,25 @@
-import { Queue, QueueRepeatMode } from "discord-player";
+import { AudioPlayerStatus } from "@discordjs/voice";
+import Queue from "../music/queue";
+import RepeatMode from "../types/repeatmode";
 
 export default function queueInfo(queue: Queue) {
-    let length = queue.tracks.reduce((x, y) => x + y.durationMS, 0);
-    const current = queue.current;
+    const tracks = queue.getTracks();
+    let length = tracks.reduce((x, y) => x + y.durationMs, 0);
```
Comment (TetroVolt) [inline src/lib/queueInfo.ts:7]: noice use of functional style programming. Way better in terms of readability than loops.

Comment (TetroVolt) [inline src/lib/queueInfo.ts:7]: noice use of functional style programming. Way better in terms of readability than loops.

File: src/music/manager.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            url: searchResult.url,
+            thumbnail: searchResult.thumbnails[0].url,
+            durationMs: searchResult.durationInSec * 1000,
+            requestedBy: user,
+        };
+
+        const track = new Track(trackInput, user);
+        return track;
+    }
+}
```
Comment (TetroVolt) [inline src/music/manager.ts:105]: Nitpick: add a new line to end of this file. Overall good use of throwing exceptions

File: src/music/queue.ts [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        try {
+            this.play(nextSong);
+            this.lastpause = Date.now();
+        } catch (e) {
+            console.error(e);
+            this.onIdle();
+        } finally {
+            this.queueLock = false;
+        }
+    };
```
Comment (TetroVolt) [inline src/music/queue.ts:286]: Normally there is the rule of thumb to limit functions to no more than 20 lines for readability but this is still pretty readable. This module is does stick to that rule, all the functions here are less than 20 and readable for the most part. 

:+1:

**Review by TetroVolt — APPROVED** [review]
Overall very nice. Readable, anything that might be long or unreadable is commented :+1: .

File: src/music/manager.ts (hunk 2 above)
Comment (valivia) [reply src/music/manager.ts]: [824cd11](https://github.com/Owl-barn/owlet/pull/3/commits/824cd114c705dd55bd95475f9e134fa4e1386a7d)
:)

**Pull request merged**